### PR TITLE
Add TempDir

### PR DESCRIPTION
**Pull request by kuba--**
Signed-off-by: kuba-- <kuba@sourced.tech>

Closes: https://github.com/src-d/go-billy/issues/54

### CODE REVIEW

File: fs.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ const (
 type Filesystem interface {
 	Basic
 	TempFile
+	TempDir
```
Comment (mcuadros) [inline fs.go]: Adding TempDir to the interface will force a major upgrade, I fill that doesn't worth to do it.
Maybe in the future with more updates.

**Comment by smola** [comment]
Can't we implement it as an utility function, just as `util.TempFile`, using the existing interfaces?

**Comment by smola** [comment]
I think :+1: to keeping it in utils without interface changes.

**Comment by kuba--** [comment]
Ok, I left only implementation in utils.

File: .gitignore [hunk 2]
```diff
@@ -1 +1,4 @@
 /coverage.txt
+/vendor
```
Comment (ajnavarro) [inline .gitignore]: should we remove that lines from .gitignore. go-billy is not using dep right now.

Comment (kuba--) [reply .gitignore]: that's why I added ;) to avoid a mistake and commit vendor dir

**Comment by kuba--** [comment]
shall we close it or merge it?

**Comment by smola** [comment]
ping @mcuadros

File: util/util.go [hunk 3]
```diff
@@ -168,6 +168,45 @@ func TempFile(fs billy.Basic, dir, prefix string) (f billy.File, err error) {
...
 }
 
+// TempDir creates a new temporary directory in the directory dir
+// with a name beginning with prefix and returns the path of the
+// new directory. If dir is the empty string, TempDir uses the
+// default directory for temporary files (see os.TempDir).
+// Multiple programs calling TempDir simultaneously
+// will not choose the same directory. It is the caller's responsibility
+// to remove the directory when no longer needed.
+func TempDir(fs billy.Dir, dir, prefix string) (name string, err error) {
```
Comment (mcuadros) [inline util/util.go]: It's not covered by any test

Comment (kuba--) [reply util/util.go]: @mcuadros - I've added 2 tests for TempDir and TempFile.

**Pull request merged**